packages/modules/viewer-react/src/types.ts

```diff
@@ -131,21 +141,53 @@ export interface ViewerInitializerParams extends ViewerIModelAppOptions {
   /** array of iTwin.js Extensions */
   extensions?: ExtensionProvider[];
 }
-
-export interface ConnectedViewerProps {
-  iTwinId: string;
+export type RequiredViewerProps = XOR<
+  XOR<ConnectedViewerProps, FileViewerProps>,
+  BlankViewerProps
+>;
+export type ViewerProps = RequiredViewerProps & ViewerCommonProps;
+export type ViewerLoaderProps = RequiredViewerProps & LoaderProps;
+export type ConnectedViewerProps = {
+  iTwinId: GuidString;
   iModelId: string;
   changeSetId?: string;
-}
-
-export interface FileViewerProps {
+};
+export type FileViewerProps = {
   /** Path to local snapshot or briefcase */
   filePath: string;
-}
+};
+
+export type BlankViewerProps = XOR<
+  XOR<
+    {
+      /** @deprecated specify location and extents instead. */
```

Comment:
thoughts on breaking these into their own types, so we can mark the whole subtype as deprecated, and not just the blankConnection property?